runtime/compilers/rillv1/parse_api.go

```diff
@@ -1,14 +1,64 @@
 package rillv1
 
 import (
+	"encoding/json"
 	"fmt"
 
+	"github.com/go-openapi/spec"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
 // APIYAML is the raw structure of a API resource defined in YAML (does not include common fields)
 type APIYAML struct {
 	DataYAML `yaml:",inline" mapstructure:",squash"`
+	OpenAPI  *OpenAPIYAML `yaml:"openapi"`
+}
+
+type OpenAPIYAML struct {
+	Info     map[string]any `yaml:"info"`
+	Request  *ReqSpecYAML   `yaml:"request"`
+	Response *RespSpecYAML  `yaml:"response"`
+}
+
+type ReqSpecYAML struct {
+	Summary    string           `yaml:"summary"`
+	Parameters []map[string]any `yaml:"parameters"`
+}
+
+type RespSpecYAML struct {
+	Description string         `yaml:"description"`
+	Schema      map[string]any `yaml:"schema"`
+}
+
+func ConvertParameters(params []map[string]any) ([]spec.Parameter, error) {
+	var parameters []spec.Parameter
+	for _, param := range params {
+		var specParam spec.Parameter
+		jsonData, err := json.Marshal(param)
+		if err != nil {
+			return nil, err
+		}
+		err = specParam.UnmarshalJSON(jsonData)
+		if err != nil {
+			return nil, err
+		}
+		parameters = append(parameters, specParam)
+	}
+	return parameters, nil
+}
+
+func ConvertSchema(schema map[string]any) (*spec.Schema, error) {
+	specSchema := spec.Schema{}
+	jsonData, err := json.Marshal(schema)
+	if err != nil {
+		return nil, err
+	}
+	err = specSchema.UnmarshalJSON(jsonData)
+	if err != nil {
+		return nil, err
+	}
+	return &specSchema, nil
 }
```

Comment:
Can we pull these into a separate OpenAPI utils package (e.g. `runtime/pkg/openapiutil`)? And maybe name them something like `MapToParameters` and `MapToSchema`. Alternatively, if kept in this package, they should be prefixed with `OpenAPI` to indicate that they're OpenAPI related (this is a big package).